### PR TITLE
Adding new authentication by token in manager

### PR DESCRIPTION
**Pull request by yangalli**
# Description

- Now we are using token based auth from the backend, instead of using firebase

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] It is urgent
- [ ] <span style="color:#d44037">Dangerous change</span>

### CODE REVIEW

File: src/services/api/userManagerApi/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { AxiosResponse } from "axios";
+import { apiPost } from "..";
+
+const userManagerApi = {
+  postUserManager: (data: any, config: any): Promise<AxiosResponse<any>> =>
```
Comment (NickNish09) [inline src/services/api/userManagerApi/index.ts:5]: I think that here would be nice to type this `data` attribute, what do you think?

File: src/pages/LoginPage/index.tsx [hunk 2]
```diff
@@ -1,34 +1,52 @@
...
     <S.Container>
       <img src={RibonIcon} alt="Ribon" />
       <S.Title>{t("title")}</S.Title>
-      <ButtonSecondary
-        text={t("buttonText")}
-        onClick={() => signInWithGoogle()}
-        leftIcon={<GoogleIcon />}
+
+      <GoogleLogin
+        clientId={process.env.REACT_APP_GOOGLE_CLIENT_ID!}
```
Comment (NickNish09) [inline src/pages/LoginPage/index.tsx:45]: Is this `!` needed here?

File: src/contexts/authenticationContext/index.tsx [hunk 3]
```diff
@@ -32,8 +26,8 @@ export const AuthenticationContext = createContext<IAuthenticationContext>(
 function AuthenticationProvider({ children }: Props) {
   const firebaseAuth = getAuth(firebaseApp);
   const navigate = useNavigate();
-  const { findOrCreateUser } = useUsers();
   const [user, setUser] = useState<User>();
+  const [accessToken, setAccessToken] = useState(localStorage.getItem("token"));
```
Comment (NickNish09) [inline src/contexts/authenticationContext/index.tsx:30]: I think that would be nice to have a `TOKEN_KEY` constant, something like:
```
export const TOKEN_KEY = 'TOKEN_KEY'
```
To be used in this localStorage calls, so it is easier to change and avoid spelling errors

File: src/contexts/authenticationContext/index.tsx [hunk 4]
```diff
@@ -42,26 +36,32 @@ function AuthenticationProvider({ children }: Props) {
...
+              Authorization: `Bearer ${response.accessToken}`,
+              "Content-Type": "application/json",
+              access_token: `${response.accessToken}`,
+            },
+          },
+        );
+
+        const token = await userManagerResponse.headers["access-token"];
+
+        localStorage.setItem("token", token);
```
Comment (NickNish09) [inline src/contexts/authenticationContext/index.tsx:55]: we would use the `TOKEN_KEY` here as well

File: src/pages/LoginPage/index.tsx (hunk 2 above)
Comment (yangalli) [reply src/pages/LoginPage/index.tsx:45]: Yes :( kkk

**Review by NickNish09 — APPROVED** [review]

**Pull request merged**